# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/ScreenOrientation.cpp
# Commit: de69e3c1635a
# Full Hash: de69e3c1635a311b5b22716b65b93fc1899a37a4
# Author: Andreas Farre <farre@mozilla.com>
# Date: 2020-08-01 09:48:10
# Regressor Bug: 1613431
# File Overlap Count: 1
# Description:
#   Bug 1613431 - Part 3: Handle synced setters return value. r=nika
#   
#   Depends on D83646
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D83647
# ==============================================================================

diff -r b326d33f98cb -r de69e3c1635a dom/base/ScreenOrientation.cpp
--- a/dom/base/ScreenOrientation.cpp	Fri Jul 31 13:37:00 2020 +0000
+++ b/dom/base/ScreenOrientation.cpp	Fri Jul 31 13:37:13 2020 +0000
@@ -79,8 +79,8 @@
 
   Document* doc = GetResponsibleDocument();
   BrowsingContext* bc = doc ? doc->GetBrowsingContext() : nullptr;
-  if (bc && !bc->InRDMPane()) {
-    bc->SetCurrentOrientation(mType, mAngle);
+  if (bc && !bc->IsDiscarded() && !bc->InRDMPane()) {
+    MOZ_ALWAYS_SUCCEEDS(bc->SetCurrentOrientation(mType, mAngle));
   }
 }
 
@@ -317,7 +317,11 @@
     return nullptr;
   }
 
-  bc->SetOrientationLock(aOrientation);
+  bc->SetOrientationLock(aOrientation, aRv);
+  if (aRv.Failed()) {
+    return nullptr;
+  }
+
   AbortInProcessOrientationPromises(bc);
   dom::ContentChild::GetSingleton()->SendAbortOtherOrientationPendingPromises(
       bc);
@@ -523,7 +527,8 @@
   }
 
   if (mType != bc->GetCurrentOrientationType()) {
-    bc->SetCurrentOrientation(mType, mAngle);
+    rv = bc->SetCurrentOrientation(mType, mAngle);
+    NS_WARNING_ASSERTION(NS_SUCCEEDED(rv), "SetCurrentOrientation failed");
 
     nsCOMPtr<nsIRunnable> runnable = DispatchChangeEventAndResolvePromise();
     rv = NS_DispatchToMainThread(runnable);
@@ -606,8 +611,10 @@
   BrowsingContext* bc = doc->GetBrowsingContext();
   if (bc && bc->GetCurrentOrientationType() !=
                 orientation->DeviceType(CallerType::System)) {
-    bc->SetCurrentOrientation(orientation->DeviceType(CallerType::System),
-                              orientation->DeviceAngle(CallerType::System));
+    nsresult result =
+        bc->SetCurrentOrientation(orientation->DeviceType(CallerType::System),
+                                  orientation->DeviceAngle(CallerType::System));
+    NS_ENSURE_SUCCESS(result, result);
 
     nsCOMPtr<nsIRunnable> runnable =
         orientation->DispatchChangeEventAndResolvePromise();